### DISCUSSION TITLE
Websocket-Connection not establishing

### DISCUSSION THREAD

**Discussion by Langohr23** [jitsi-contrib/jitsi-helm#9] [Q&A]
Hello,

I setup the jitsi over helm charts an can join the session. Sadly it is not connecting for the websocket-Connection. Any Ideas how to fix?

`strophe.umd.js:5463 WebSocket connection to 'wss://jitsi.myurl.com/xmpp-websocket?room=wrongmindsmaximizeuneasily' failed: 
_connect @ strophe.umd.js:5463
connect @ strophe.umd.js:2368
_interceptConnectArgs @ strophe.stream-management.js:224
connect @ XmppConnection.js:242
_connect @ xmpp.js:414
connect @ xmpp.js:486
c.connect @ JitsiConnection.js:61
e @ connection.js:47
(anonymous) @ connection.js:178
d @ connection.js:103
h @ connection.js:208
W @ conference.js:185
createInitialLocalTracksAndConnect @ conference.js:654
init @ conference.js:776
Logger.js:154 2021-06-29T17:13:39.187Z [JitsiMeetJS.js] <Object.getGlobalOnErrorHandler>:  UnhandledError: Strophe: Websocket error [object Event] Script: null Line: null Column: null StackTrace:  Error: Strophe: Websocket error [object Event]
    at Object.r.Strophe.log (https://jitsi.myurl.com/libs/lib-jitsi-meet.min.js?v=4628:17:16531)
    at Object.error (https://jitsi.myurl.com/libs/lib-jitsi-meet.min.js?v=4628:1:23431)
    at N.Websocket._onError (https://jitsi.myurl.com/libs/lib-jitsi-meet.min.js?v=4628:1:62905)
o @ Logger.js:154
getGlobalOnErrorHandler @ JitsiMeetJS.js:613
window.onerror @ middleware.js:110
callErrorHandler @ GlobalOnErrorHandler.js:61
r.Strophe.log @ strophe.util.js:89
error @ strophe.umd.js:1392
_onError @ strophe.umd.js:5730
error (async)
_connect @ strophe.umd.js:5465
connect @ strophe.umd.js:2368
_interceptConnectArgs @ strophe.stream-management.js:224
connect @ XmppConnection.js:242
_connect @ xmpp.js:414
connect @ xmpp.js:486
c.connect @ JitsiConnection.js:61
e @ connection.js:47
(anonymous) @ connection.js:178
d @ connection.js:103
h @ connection.js:208
W @ conference.js:185
createInitialLocalTracksAndConnect @ conference.js:654
init @ conference.js:776
async function (async)
init @ conference.js:741
(anonymous) @ actions.web.js:31
Promise.then (async)
(anonymous) @ actions.web.js:30
(anonymous) @ index.js:11
(anonymous) @ middleware.js:29
(anonymous) @ middleware.js:32
(anonymous) @ middleware.js:31
(anonymous) @ middleware.web.js:33
(anonymous) @ middleware.any.js:22
(anonymous) @ middleware.js:67
(anonymous) @ middleware.js:43
(anonymous) @ middleware.js:61
(anonymous) @ middleware.js:79
(anonymous) @ middleware.js:71
(anonymous) @ middleware.js:77
(anonymous) @ middleware.js:39
(anonymous) @ middleware.js:106
(anonymous) @ middleware.js:38
(anonymous) @ middleware.js:33
(anonymous) @ middleware.web.js:24
(anonymous) @ middleware.any.js:93
(anonymous) @ middleware.js:77
(anonymous) @ middleware.web.js:21
(anonymous) @ middleware.js:44
(anonymous) @ middleware.js:25
(anonymous) @ middleware.js:16
(anonymous) @ middleware.js:21
(anonymous) @ middleware.js:23
(anonymous) @ middleware.js:21
(anonymous) @ middleware.js:111
(anonymous) @ middleware.js:16
(anonymous) @ middleware.js:42
(anonymous) @ middleware.js:33
(anonymous) @ middleware.js:176
(anonymous) @ middleware.js:39
(anonymous) @ middleware.js:24
(anonymous) @ middleware.js:26
(anonymous) @ middleware.web.js:20
(anonymous) @ middleware.js:42
(anonymous) @ middleware.js:148
(anonymous) @ middleware.js:21
(anonymous) @ middleware.js:60
(anonymous) @ middleware.js:44
(anonymous) @ middleware.js:49
(anonymous) @ middleware.js:28
(anonymous) @ middleware.js:12
(anonymous) @ middleware.js:39
(anonymous) @ middleware.js:31
(anonymous) @ middleware.web.js:22
(anonymous) @ middleware.any.js:100
(anonymous) @ middleware.js:106
(anonymous) @ middleware.js:19
(anonymous) @ middleware.js:29
(anonymous) @ middleware.js:91
(anonymous) @ middleware.js:101
(anonymous) @ middleware.js:24
(anonymous) @ middleware.js:20
(anonymous) @ middleware.js:15
(anonymous) @ middleware.js:23
(anonymous) @ middleware.js:20
(anonymous) @ middleware.js:25
(anonymous) @ middleware.js:74
(anonymous) @ middleware.js:40
(anonymous) @ middleware.js:158
_start @ Conference.js:250
componentDidMount @ Conference.js:133
js @ react-dom.production.min.js:238
t.unstable_runWithPriority @ scheduler.production.min.js:20
pa @ react-dom.production.min.js:113
Es @ react-dom.production.min.js:230
ms @ react-dom.production.min.js:206
(anonymous) @ react-dom.production.min.js:114
t.unstable_runWithPriority @ scheduler.production.min.js:20
pa @ react-dom.production.min.js:113
ba @ react-dom.production.min.js:114
ga @ react-dom.production.min.js:113
ds @ react-dom.production.min.js:204
enqueueSetState @ react-dom.production.min.js:127
_.setState @ react.production.min.js:13
(anonymous) @ BaseApp.js:241
_navigate @ BaseApp.js:240
(anonymous) @ middleware.js:149
Promise.then (async)
(anonymous) @ middleware.js:149
(anonymous) @ middleware.js:169
(anonymous) @ middleware.js:26
(anonymous) @ middleware.js:32
(anonymous) @ middleware.js:31
(anonymous) @ middleware.web.js:33
(anonymous) @ middleware.any.js:22
(anonymous) @ middleware.js:67
(anonymous) @ middleware.js:43
(anonymous) @ middleware.js:61
(anonymous) @ middleware.js:79
(anonymous) @ middleware.js:71
(anonymous) @ middleware.js:77
(anonymous) @ middleware.js:147
(anonymous) @ middleware.js:35
(anonymous) @ middleware.js:106
(anonymous) @ middleware.js:38
(anonymous) @ middleware.js:33
(anonymous) @ middleware.web.js:24
(anonymous) @ middleware.any.js:93
(anonymous) @ middleware.js:77
(anonymous) @ middleware.web.js:21
(anonymous) @ middleware.js:44
(anonymous) @ middleware.js:25
(anonymous) @ middleware.js:16
(anonymous) @ middleware.js:21
(anonymous) @ middleware.js:23
(anonymous) @ middleware.js:21
(anonymous) @ middleware.js:111
(anonymous) @ middleware.js:16
(anonymous) @ middleware.js:42
(anonymous) @ middleware.js:33
(anonymous) @ middleware.js:176
(anonymous) @ middleware.js:39
(anonymous) @ middleware.js:24
(anonymous) @ middleware.js:26
(anonymous) @ middleware.web.js:20
(anonymous) @ middleware.js:42
(anonymous) @ middleware.js:148
(anonymous) @ middleware.js:21
(anonymous) @ middleware.js:197
(anonymous) @ middleware.js:45
(anonymous) @ middleware.js:44
(anonymous) @ middleware.js:49
(anonymous) @ middleware.js:28
(anonymous) @ middleware.js:12
(anonymous) @ middleware.js:39
(anonymous) @ middleware.js:31
(anonymous) @ middleware.web.js:22
(anonymous) @ middleware.any.js:446
(anonymous) @ middleware.any.js:93
(anonymous) @ middleware.js:96
(anonymous) @ middleware.js:19
(anonymous) @ middleware.js:29
(anonymous) @ middleware.js:91
(anonymous) @ middleware.js:101
(anonymous) @ middleware.js:24
(anonymous) @ middleware.js:20
(anonymous) @ middleware.js:15
(anonymous) @ middleware.js:23
(anonymous) @ middleware.js:20
(anonymous) @ middleware.js:25
(anonymous) @ middleware.js:74
(anonymous) @ middleware.js:40
(anonymous) @ middleware.js:158
dispatch @ redux.js:636
(anonymous) @ actions.js:138
async function (async)
(anonymous) @ actions.js:111
(anonymous) @ index.js:11
(anonymous) @ middleware.js:29
(anonymous) @ middleware.js:32
(anonymous) @ middleware.js:31
(anonymous) @ middleware.web.js:33
(anonymous) @ middleware.any.js:22
(anonymous) @ middleware.js:67
(anonymous) @ middleware.js:43
(anonymous) @ middleware.js:61
(anonymous) @ middleware.js:79
(anonymous) @ middleware.js:71
(anonymous) @ middleware.js:77
(anonymous) @ middleware.js:39
(anonymous) @ middleware.js:106
(anonymous) @ middleware.js:38
(anonymous) @ middleware.js:33
(anonymous) @ middleware.web.js:24
(anonymous) @ middleware.any.js:93
(anonymous) @ middleware.js:77
(anonymous) @ middleware.web.js:21
(anonymous) @ middleware.js:44
(anonymous) @ middleware.js:25
(anonymous) @ middleware.js:16
(anonymous) @ middleware.js:21
(anonymous) @ middleware.js:23
(anonymous) @ middleware.js:21
(anonymous) @ middleware.js:111
(anonymous) @ middleware.js:16
(anonymous) @ middleware.js:42
(anonymous) @ middleware.js:33
(anonymous) @ middleware.js:176
(anonymous) @ middleware.js:39
(anonymous) @ middleware.js:24
(anonymous) @ middleware.js:26
(anonymous) @ middleware.web.js:20
(anonymous) @ middleware.js:42
(anonymous) @ middleware.js:148
(anonymous) @ middleware.js:21
(anonymous) @ middleware.js:60
(anonymous) @ middleware.js:44
(anonymous) @ middleware.js:49
(anonymous) @ middleware.js:28
(anonymous) @ middleware.js:12
(anonymous) @ middleware.js:39
(anonymous) @ middleware.js:31
(anonymous) @ middleware.web.js:22
(anonymous) @ middleware.any.js:100
(anonymous) @ middleware.js:106
(anonymous) @ middleware.js:19
(anonymous) @ middleware.js:29
(anonymous) @ middleware.js:91
(anonymous) @ middleware.js:101
(anonymous) @ middleware.js:24
(anonymous) @ middleware.js:20
(anonymous) @ middleware.js:15
(anonymous) @ middleware.js:23
(anonymous) @ middleware.js:20
(anonymous) @ middleware.js:25
(anonymous) @ middleware.js:74
(anonymous) @ middleware.js:40
(anonymous) @ middleware.js:158
_openURL @ AbstractApp.js:113
(anonymous) @ AbstractApp.js:47
Promise.then (async)
componentDidMount @ AbstractApp.js:44
js @ react-dom.production.min.js:238
t.unstable_runWithPriority @ scheduler.production.min.js:20
pa @ react-dom.production.min.js:113
Es @ react-dom.production.min.js:230
ds @ react-dom.production.min.js:204
Vs @ react-dom.production.min.js:263
Ws @ react-dom.production.min.js:263
(anonymous) @ react-dom.production.min.js:272
ys @ react-dom.production.min.js:208
el @ react-dom.production.min.js:272
render @ react-dom.production.min.js:273
Rs.renderEntryPoint @ index.web.js:71
(anonymous) @ WrongMindsMaximizeUneasily:25
Show 204 more frames
Logger.js:154 2021-06-29T17:13:39.189Z [modules/xmpp/strophe.util.js] <Object.r.Strophe.log>:  Strophe: Websocket error [object Event]`

Thank you!

**Comment by Langohr23** [comment]
Her the same error with a screenshot of the browser-console:
![image](https://user-images.githubusercontent.com/33636936/125174475-13639680-e1c6-11eb-8200-25064f829206.png)

**Comment by vanducvo** [comment]
Hi, I think this error raise for prosody don't accept cross domain.
https://github.com/jitsi-contrib/jitsi-helm/blob/d504428bad63ec839bec0b666d82b97dbfff66fa/values.yaml#L196

replace by:
```yaml
extraCommonEnvs:
  XMPP_CROSS_DOMAIN: "true"
```

**Comment by ThomasFen** [comment]
same here, this chart doesn't work out of the box with just setting the public ip. It's not the XMPP_CROSS_DOMAIN at fault - in that case it would return 403 .

 ` extraEnvs: {ENABLE_XMPP_WEBSOCKET: "0"}` in the web section of values.yaml fixed it for me

**Comment by spijet** [comment]
This might also be caused by the `web` container being unable to resolve the other containers' names properly. The best solution here is to set your DNS service's IP address in [`.Values.web.resolverIP`](https://github.com/jitsi-contrib/jitsi-helm/blob/main/values.yaml#L56).
Disabling XMPP websockets (and using HTTP binds instead) should work too, but that's not a supported configuration for Jitsi Meet nowadays, IIRC.
